### PR TITLE
[azquery] migration guide

### PR DESCRIPTION
**Pull request by gracewilcox**
Added migration guide for azquery module.

### CODE REVIEW

File: sdk/monitor/azquery/MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+The new `azquery` module condenses metrics and logs functionality into one package for simpler access. The `azquery` module contains two clients: LogsClient and MetricsClient.
+
+Transitioning to a single package has resulted in a number of name changes, as detailed below.
+
+### Log name changes
+
+| `operationalinsights`    | `azquery` |
+| ----------- | ----------- |
+| QueryClient.Execute      | LogsClient.QueryWorkspace     |
+| MetadataClient.Get and MetadataClient.Post | N/A |
```
[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/monitor/azquery/MIGRATION.md:20]: Is there anything users can do with `MetadataClient` that they can't do with `azquery`?

[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md:20]: `MetadataClient` allows users to query metadata from Logs data. No current plans to add the functionality.

File: sdk/monitor/azquery/MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+This guide is intended to assist in the migration to the `azquery` module. `azquery` allows users to retrieve log and metric data from Azure Monitor.
+
+## Package consolidation
+
+ Azure Monitor allows users to retrieve telemetry data for their Azure resources. The main two data catagories for Azure Monitor are [metrics](https://learn.microsoft.com/azure/azure-monitor/essentials/data-platform-metrics) and [logs](https://learn.microsoft.com/azure/azure-monitor/logs/data-platform-logs). 
+ 
+ There have been a number of [terminology](https://learn.microsoft.com/azure/azure-monitor/terminology) changes for Azure Monitor over the years which resulted in the operations being spread over multiple packages. For Go, metrics methods were contained in `github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/<version-number>/insights` and logs methods resided in `github.com/Azure/azure-sdk-for-go/services/operationalinsights/v1/operationalinsights`.
+
+The new `azquery` module condenses metrics and logs functionality into one package for simpler access. The `azquery` module contains two clients: LogsClient and MetricsClient.
```
Comment (richardpark-msft) [inline sdk/monitor/azquery/MIGRATION.md:11]: Good opportunity to link to pkg.go.dev resources for these clients.

File: sdk/monitor/azquery/MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    res, err := client.Execute(context.TODO(), "DEMO_WORKSPACE", operationalinsights.QueryBody{Query: &query, Timespan: &timespan})
+	if err != nil {
+		//TODO: handle error
+	}
+```
+
+### `azquery`
+
+Compared to previous versions, querying logs with the new `azquery` module is clearer and simpler. There are a number of name changes for clarity, like how the old `Execute` method is now `QueryWorkspace`. In addition, there is improved time support. Before if a user added a timespan over which to query the request, it had to be a string constructed in the ISO8601 interval format. Users frequently made mistakes when constructing this string. With the new `QueryWorkspace` method, the type of timespan has been changed from a string to a new type named `TimeInterval`. `TimeInterval` has a contructor that allows users to take advantage of Go's time package, allowing easier creation.
```
Comment (richardpark-msft) [inline sdk/monitor/azquery/MIGRATION.md:61]: I think you can do some quick pkg.go.dev links here too for QueryWorkspace and TimeInterval.

File: sdk/monitor/azquery/MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+There are also a number of changes in the shape of the response body to be mindful of.
```
Comment (richardpark-msft) [inline sdk/monitor/azquery/MIGRATION.md:63]: You can't give me an ominous warning like that and not show me what you mean....

File: sdk/monitor/azquery/MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+Transitioning to a single package has resulted in a number of name changes, as detailed below.
+
+### Log name changes
+
+| `operationalinsights`    | `azquery` |
+| ----------- | ----------- |
+| QueryClient.Execute      | LogsClient.QueryWorkspace     |
+| MetadataClient.Get and MetadataClient.Post | N/A |
+
+The `azquery` module does not contain the `MetadataClient`. For that functionality, please use the old `operationalinsights` module or [file an issue in our github repo](https://github.com/Azure/azure-sdk-for-go/issues), so we can prioritize adding it to `azquery`.
```
Comment (richardpark-msft) [inline sdk/monitor/azquery/MIGRATION.md:22]: Probably worth providing the pkg.go.dev link to the operationalinsights package.

**Review by richardpark-msft — APPROVED** [review]
Linking to types makes it a bit simpler, but no major issues. Approving pending resolved feedback.

File: sdk/monitor/azquery/MIGRATION.md (hunk 5 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md]: fixed in 36b33f8

File: sdk/monitor/azquery/MIGRATION.md (hunk 4 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md]: fixed in 36b33f8

File: sdk/monitor/azquery/MIGRATION.md (hunk 3 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md]: fixed in 36b33f8

File: sdk/monitor/azquery/MIGRATION.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md]: fixed in 36b33f8

File: sdk/monitor/azquery/MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+| ----------- | ----------- |
+| MetricsClient.List     | MetricsClient.QueryResource       |
+| MetricDefinitionsClient.List   | MetricsClient.NewListDefinitionsPager        |
+| MetricNamespacesClient.List   | MetricsClient.NewListNamespacesPager        |
+
+## Query Logs
+
+### `operationalinsights`
+```go
+    import (
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/azquery/MIGRATION.md]: fixed in 36b33f8

**Pull request merged**